File: 45-JumpGameII/solution.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int sz = nums.size();
        vector<int> step(sz, 0);
        int last = 0;
        for(int i = 0; i < sz; i++){
            for(int j = last+1; j < min(i+nums[i]+1, sz); j++)
                step[j] = step[i]+1;
            last = max(last, i+nums[i]);
        }
        return step[sz-1];
    }
};

File: 45-JumpGameII/solution.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sz = len(nums)
        last, step = 0, [0]*sz
        for i in xrange(sz):
            for j in xrange(last+1, min(i+nums[i]+1, sz)):
                step[j] = step[i]+1
            last = max(last, i+nums[i])
        return step[sz-1]
